portfolio/src/main/webapp/sheetsApi.js

```diff
@@ -113,39 +118,109 @@ function handleAuthorizationError(error) {
+  scheduledTaskAsSheetsArray.push(taskDescription);
+  scheduledTaskAsSheetsArray.push(taskPriority);
+
+  return scheduledTaskAsSheetsArray;
+}
+
 /**
  * Creates and populates a new spreadsheet with the scheduled task information.
  * Also displays the link to the spreadsheet as text on the UI.
  */
 function handleExportSchedule() {
+  const scheduledTaskCount = scheduledTasks.length;
+  if (scheduledTaskCount == 0) {
+    $('#sheets-message')
+        .removeClass('d-none')
+        .text('Cannot export empty schedule.')
+        .show();
+    return;
+  }
+
   var spreadsheetProperties = {
     title: 'Your Scheduled Tasks'
-    // TODO(tomasalvarez): Add desired creation properties.
-  };
-
-  var spreadsheetBody = {
-      // TODO(tomasalvarez): Add desired properties to the request body.
+    // TODO(tomasalvarez): Include suggestions of adding the creation time
+    //     in the title of the spreadsheet.
   };
 
   var request = gapi.client.sheets.spreadsheets.create(
       {properties: spreadsheetProperties});
-  request.then(
-      function(response) {
-        // TODO(tomasalvarez): Change code below to process the `response` 
-        // object. Currently just displays the link back to the user.
-        $('#sheets-url-container').removeClass('d-none');
-        $('#sheets-url-container')
-            .text(
-                'Done. Here is the URL to your spreadsheet: ' +
-                response.result.spreadsheetUrl);
-      },
-      function(reason) {
-        handleExportError(reason);
-      });
+  request.then((response) => {
+    var values = [[
+      'Task', 'Scheduled Time', 'Duration (minutes)', 'Description', 'Priority'
+    ]];
+    // This is using the new scheduledTasks variable which is a local copy
+    // of the latest results from scheduling.
+    makeSheetsValuesFromScheduledTasks(scheduledTasks, values);
+    // This is a blank row in the spreadsheet so you can see the totals easier.
+    values.push([]);
+    values.push([
+      scheduledTaskCount.toString() + ' Tasks Total', '',
+      '=SUM(C2:C' + (scheduledTaskCount + 1).toString() +
+          ') & " Total Minutes Scheduled"'
+    ])
+
+    var data = {
+      // This is +3 because we have the initial row with titles and the final
+      // rows with some totals.
+      'range': 'Sheet1!A1:E' + (scheduledTaskCount + 3).toString(),
+      'majorDimension': 'ROWS',
+      'values': values
+    }  // Additional ranges to update.
+
+    var body = {data: data, valueInputOption: 'USER_ENTERED'};
+    gapi.client.sheets.spreadsheets.values
+        .batchUpdate(
+            {spreadsheetId: response.result.spreadsheetId, resource: body})
+        .then(
+            function(response) {
+              //  This displays the link back to the user.
+              $('#sheets-url-container').removeClass('d-none');
+              $('#sheets-url-container')
+                  .text(
+                      'Here is the URL to your spreadsheet: ' +
```

Comment:
Can we make this an a tag with link, rather than just text?

Reply:
Done.